Implementation of import for CPT. Fixes #2307

stevekogo opened the pull request.

stevekogo commented:
@africanmathsinitiative/instat This branch task is ready for review.

lilyclements commented on instat/dlgExportToCPT.vb:
```diff
@@ -25,17 +27,152 @@ Public Class dlgExportToCPT
+        txtExportFile.Text = ""
+        chkLong.Checked = True
+        ucrReceiverYears.SetMeAsReceiver()
+        ucrSSTDataframe.Reset()
```
We also want as a default for ucrLocationDataFrame to reset

lilyclements commented on instat/dlgExportToCPT.vb:
```diff
@@ -25,17 +27,152 @@ Public Class dlgExportToCPT
+    End Sub
+
+    Private Sub chkLong_CheckedChanged(sender As Object, e As EventArgs) Handles chkLong.CheckedChanged
+        If chkLong.Checked Then
```
Just as a small thing here, if we have say "Stations:" selected, then check "Long", everything changes great, however, "Stations" is still selected so the user has to reclick in the "Year" or "Data Column" and then select their variable.
It would be good is when chkLong is checked and unchecked, for the ucrYear to become the focus?

lilyclements commented on instat/dlgExportToCPT.Designer.vb:
```diff
@@ -22,32 +22,222 @@ Partial Class dlgExportToCPT
+        Me.chkLong.TabIndex = 0
+        Me.chkLong.Tag = "Long"
+        Me.chkLong.Text = "Long"
+        Me.chkLong.UseVisualStyleBackColor = True
```
What does "Long" do out of interest? I can't tell from the name.
Additionally, what are the advantages to having group box with only one item in it as oppose to the one checkbox by itself?

lilyclements commented on instat/dlgExportToCPT.Designer.vb:
```diff
@@ -22,32 +22,222 @@ Partial Class dlgExportToCPT
+        Me.grpDataFormat.Controls.Add(Me.chkLong)
+        Me.grpDataFormat.Location = New System.Drawing.Point(315, 16)
+        Me.grpDataFormat.Name = "grpDataFormat"
+        Me.grpDataFormat.Size = New System.Drawing.Size(82, 42)
```
I think we want the group box to be the same size as the ucr receivers. That is, (120, 42) as the size.
Then we would want the location to be in line with the receivers, so at about (296, 16)

lilyclements commented on instat/dlgExportToCPT.Designer.vb:
```diff
@@ -22,32 +22,222 @@ Partial Class dlgExportToCPT
+        Me.lblYears.Size = New System.Drawing.Size(34, 13)
+        Me.lblYears.TabIndex = 2
+        Me.lblYears.Tag = "Years"
+        Me.lblYears.Text = "Years"
```
For all four labels, we want their location to be at about 296 to be consistent with other dialogs.
Additionally, they need to have colons

stevekogo commented:
@lilyclements Thanks for the comments. I have implemented them apart from positioning of ucrLocationDataFrame which I think will make the dialog less appealing if it in line with the receivers as it will create a lot of space. I welcome more reviews on what I have implemented.

stevekogo commented:
@dannyparsons have a look at this. I think it is ready to go in.
